packages/core/lib/match.js

```diff
@@ -21,6 +21,10 @@ function match (test, options, configSetters) {
 
   assertConfigSetters(configSetters)
 
+  if (options.test) {
+    console.warn(`match(): Setting 'test' in options is not supported and will be overriden with a 'test' argument.`)
```

Comment:
Maybe we should even throw! What do you think?

Seems useful to defuse the bomb instead of putting a warning sign on it.